### ISSUE TITLE
Castlevania Harmony of Despair save game hanging. & DLC character duplication.

### ISSUE THREAD

**Issue by sonikku** [RPCS3/rpcs3#13262]
### Quick summary

When starting the game for the first time the game hangs for more than 5 minutes while generating a save file. Even when loading the save sometimes it hangs. When that occurs I have to clear all the cache and do the process all over again.

Also there is a problem with DLC characters duplicating themselves and causing the emulator to crash. The community does have a hacky way around this but I'd prefer if this bug was addressed as well.

### Attach a log file

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10471149/RPCS3.log.gz)

**Comment by Xeeynamo** [comment]
I have the same issue. Following.

**Comment by Rex000** [comment]
Same issue here as well. Weird that nobody attended to it.

**Comment by Megamouse** [comment]
The game isn't really stuck loading the save game.
It actually keeps compiling a ton of PPU modules, which you can currently only see in the log.
This is especially unfortunate if you have a really old and slow CPU.

You can try with the PPU Interpreter instead. It may help you go past that loading screen faster, but also may end up slowing the game down in general.

**Comment by Rex000** [comment]
Thanks it does get past that screen now but the options(?) screen immediately after it shows nothing but the background.
![NPUB30505_screenshot_2023_02_21_23_45_47](https://user-images.githubusercontent.com/28770012/220433708-309573c0-c231-4514-b44b-a25f3b3c2012.png)

**Comment by Rex000** [comment]
I actually had Write color buffers on before but it didn't work for some reason. It worked when I closed and started the emulator again. The game runs at under 30 fps, though the screen stays like this the whole time instead of zooming in on the character.

![NPUB30505_screenshot_2023_02_22_01_20_39](https://user-images.githubusercontent.com/28770012/220450339-39341ffc-1aba-48e6-89c8-f0e1e0680c61.png)

**Comment by Rex000** [comment]
Write color buffers was enabled but it didn't work. I restarted the emulator then it worked.

**Comment by sonikku** [comment]
This problem still exists even now. Any progress on the endless PPU Looping for this game?

**Comment by saechaol** [comment]
> This problem still exists even now. Any progress on the endless PPU Looping for this game?

It is neither endless nor is it looping. If you look at the console log (View > Show Log/TTY), you can see that RPCS3 is compiling the PPU modules in sequence 1 at a time; at each step it prints its progress to the console. It’s not that _nothing_ is happening, it’s that it just takes 10 minutes or so to complete, assuming decent CPU.

**Comment by Megamouse** [comment]
The only bug here is the DLC thing if it is still an issue

**Comment by saechaol** [comment]
> The only bug here is the DLC thing if it is still an issue

Yeah, I edited my last comment to reflect this; it’s in my opinion that OP rewrites this issue to be scoped specifically to address the duplication error. The experienced hanging seems to be the compiler working as intended, even if it’s not immediately apparent to the user.

**Comment by ghost** [comment]
Can we get an update on this?

**Comment by ScottNash042** [comment]
RPCS3 v0.0.40-19235-a088b18e
just tried this out
the DLC character duplication doesnt happen any more
when you first load the game it still takes a while to load
keeps on coming up with the error
{PPU[0x1000004] Thread () [HLE:0x00d11494, LR:0x007be4c4]} SYS: 'sceNpDrmIsAvailable' failed with 0x80029513 : SCE_NP_DRM_ERROR_NO_ENTITLEMENT

[RPCS3.log.gz](https://github.com/user-attachments/files/26923186/RPCS3.log.gz)
